modules/javafx.controls/src/test/java/test/javafx/scene/control/TableViewTest.java

```diff
@@ -352,6 +352,50 @@
         assertEquals("Item 2", fm.getFocusedItem());
     }
 
+    @Test
+    public void ensureRowRemainsSelectedWhenSelectingCellInSameRow() {
+        class Person {
+            final String firstName, lastName;
+            Person(String firstName, String lastName) {
+                this.firstName = firstName;
+                this.lastName = lastName;
+            }
+            public String getFirstName() { return firstName; }
+            public String getLastName() { return lastName; }
+        }
+
+        var tableView = new TableView<Person>();
+        tableView.getSelectionModel().setCellSelectionEnabled(true);
+        tableView.getSelectionModel().setSelectionMode(SelectionMode.MULTIPLE);
+
+        var col1 = new TableColumn<Person, String>();
+        col1.setCellValueFactory(new PropertyValueFactory<>("firstName"));
+        var col2 = new TableColumn<Person, String>();
+        col2.setCellValueFactory(new PropertyValueFactory<>("lastName"));
+        tableView.getColumns().addAll(List.of(col1, col2));
+
+        var ab = new Person("a", "b");
+        tableView.getItems().add(ab);
+        var cd = new Person("c", "d");
+        tableView.getItems().add(cd);
+
+        var selectionModel = tableView.getSelectionModel();
+        selectionModel.select(0);
+        selectionModel.clearAndSelect(0, col1);
+
+        // The following asserts should work once JDK-8273336 is fixed:
```

Comment:
@mstr2 About this issue, I've added a [comment](https://bugs.openjdk.java.net/browse/JDK-8273336?focusedCommentId=14445984&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-14445984), in case you have some time to check it. 
With my possible fix, these assertions work fine, but fail without it.